R/visualization-functions.R

```diff
@@ -654,69 +654,192 @@ boxplot_vars <- function(md, include_vars, x_var) {
 #'
 #' @inheritParams collapse_duplicate_hgnc_symbol
 #' @inheritParams filter_genes
+#' @inheritParams identify_outliers
 #' @param sex_var Column name of the sex or gender-specific metadata.
 #' @export
-plot_sexcheck <- function(clean_metadata, count_df, biomart_results, sex_var) {
-  md <- tibble::rownames_to_column(clean_metadata, var = "sampleId") %>%
-    dplyr::select(.data$sampleId, !!sex_var)
-  sex_specific <- count_df[
-    grepl(
-      paste0(
-        rownames(
-          biomart_results[
-            biomart_results$hgnc_symbol %in% c("UTY", "XIST"),
-            ]),
-        collapse = "|"
-        ),
-      rownames(count_df)
-      ),
-    ]
-  rownames(sex_specific) <- convert_geneids(sex_specific)
-  sex_specific <- tibble::rownames_to_column(sex_specific, var = "geneId")
-  results <- dplyr::select(
-    biomart_results,
-    .data$hgnc_symbol,
-    .data$chromosome_name
-    )
-  results <- dplyr::filter(results, .data$hgnc_symbol %in% c("XIST", "UTY"))
-  results <- tibble::rownames_to_column(results, var = "geneId")
-  results <- dplyr::left_join(results, sex_specific)
-  results <- tidyr::pivot_longer(
-    results,
-    -dplyr::all_of(c("geneId", "chromosome_name", "hgnc_symbol")),
-    names_to = "sampleId",
-    values_to = "counts(log)"
-    ) %>%
-    dplyr::mutate(
-      `counts(log)` = log(.data$`counts(log)`),
-      `counts(log)` = ifelse(
-        .data$`counts(log)` == -Inf, 0, .data$`counts(log)`
+plot_sexcheck <- function(clean_metadata, filtered_counts, biomart_results, sex_var) {
```

Comment:
Need XIST and UTY expression to correlate PCs to sex-specific expression. Take `count_df` as input instead of `filtered_counts`. Pass `count_df` to `filter_genes`, but preserve XIST, UTY expression explicitly.

Reply:
We are in agreement to not change `plot_sexcheck` in this PR.